src/Microsoft.DotNet.Build.Tasks/PackageFiles/notsupported.targets

```diff
@@ -1,15 +1,14 @@
 <?xml version="1.0" encoding="utf-8"?>
 <Project ToolsVersion="12.0" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
 
-  <PropertyGroup Condition="'$(GeneratePlatformNotSupportedAssembly)' == 'true'">
```

Comment:
I wouldn't force people to set the string IMHO.  We have plenty of cases where we don't use it.  If I were you I would create a single parameter: -throw.  If a value is specified then use it as the message, if no value is specified then don't pass a message and use the default.  My care level about SR vs quotes is low, but I think you're just creating more work for yourself by changing the behavior here.